sbi/inference/snpe/snpe_base.py

```diff
@@ -240,12 +243,13 @@ def __call__(
 
     @abstractmethod
     def _log_prob_proposal_posterior(
-        self, theta: Tensor, x: Tensor, masks: Tensor, **kwargs
+        self, theta: Tensor, x: Tensor, masks: Tensor, proposal: DirectPosterior
     ) -> Tensor:
         raise NotImplementedError
 
     def _train(
         self,
+        proposal: Any,
```

Comment:
It can indeed be `Any`. I also made it `Any` for `_log_prob_proposal_posterior` in snpe_c now.

The reason is that: if we use e.g. atomic snpe-c, we do not need to evaluate the proposal. All we need from it is to draw samples, which we then pass to the simulator. Any distribution can do this.

Only as soon as we call non-atomic loss, it has to be a `DirectPosterior`. Therefore, in `_log_prob_proposal_posterior_mog`, the type of `proposal` is `DirectPosterior`.